axum-extra/src/extract/cookie/mod.rs

```diff
@@ -280,6 +283,24 @@ mod tests {
     cookie_test!(private_cookies, PrivateCookieJar);
     cookie_test!(private_cookies_with_custom_key, PrivateCookieJar<CustomKey>);
 
+    #[derive(Clone)]
+    struct AppState {
+        key: Key,
+        custom_key: CustomKey,
+    }
+
+    impl From<AppState> for Key {
+        fn from(state: AppState) -> Key {
+            state.key
+        }
+    }
```

Comment:
So, what does the compiler error look like when this impl is missing?

Reply:
Its pretty good actually

```
error[E0277]: the trait bound `cookie::Key: From<AppState>` is not satisfied
   --> axum-extra/src/extract/cookie/mod.rs:236:36
    |
236 |                     .route("/set", get(set_cookie))
    |                                    ^^^ the trait `From<AppState>` is not implemented for `cookie::Key`
...
281 |     cookie_test!(signed_cookies, SignedCookieJar);
    |     --------------------------------------------- in this macro invocation
    |
    = help: the trait `From<CustomKey>` is implemented for `cookie::Key`
    = note: required because of the requirements on the impl of `Into<cookie::Key>` for `AppState`
note: required because of the requirements on the impl of `axum::extract::FromRequest<hyper::Body, AppState>` for `signed::SignedCookieJar`
```